### PR TITLE
[Testing] Fix core tests

### PR DESCRIPTION
**Pull request by cadedaniel**
https://github.com/vllm-project/vllm/commit/24aecf421a4ad5989697010963074904fead9a1b started causing failures after it was merged.

### CODE REVIEW

File: vllm/sequence.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def __init__(
         prompt: str,
         prompt_token_ids: List[int],
         block_size: int,
-        eos_token_id: int,
+        eos_token_id: Optional[int] = None,
```
Comment (cadedaniel) [inline vllm/sequence.py:145]: cc @njhill FYI https://github.com/vllm-project/vllm/pull/3166

**Comment by grandiose-pizza** [comment]
Thanks for this. My tests were failing:
https://github.com/vllm-project/vllm/pull/3183

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thanks for the quick fix!

File: vllm/sequence.py (hunk 1 above)
Comment (njhill) [reply vllm/sequence.py:145]: Thanks @cadedaniel and apologies for missing these tests in the original PR.

**Pull request merged**